### PR TITLE
[Reply] Use NavigationSuite for nav UI components

### PR DESCRIPTION
**Pull request by jdkoren**

### CODE REVIEW

File: Reply/app/src/main/java/com/example/reply/ui/navigation/ReplyNavigationComponents.kt [hunk 1]
```diff
@@ -49,20 +50,119 @@ import androidx.compose.material3.NavigationRail
...
+        WindowHeightSizeClass.EXPANDED -> ReplyNavigationContentPosition.CENTER
+        else -> ReplyNavigationContentPosition.TOP
+    }
+
+    val drawerState = rememberDrawerState(initialValue = DrawerValue.Closed)
+    val coroutineScope = rememberCoroutineScope()
+    // Avoid opening the modal drawer when there is a permanent drawer or a bottom nav bar,
+    // but always allow closing an open drawer.
+    val gesturesEnabled =
+        drawerState.isOpen || navLayoutType == NavigationSuiteType.NavigationRail
```
Comment (alexvanyo) [inline Reply/app/src/main/java/com/example/reply/ui/navigation/ReplyNavigationComponents.kt:118]: This logic does makes sense for consistency, where the drawer will remain open if it was open, but maybe an alternative would be to just hide it immediately if the navigation layout type isn't a navigation rail? Not sure which way would be better.

**Review by alexvanyo — APPROVED** [review]

File: Reply/gradle/libs.versions.toml [hunk 2]
```diff
@@ -10,20 +10,25 @@ androidx-appcompat = "1.6.1"
 androidx-benchmark = "1.2.0"
 androidx-benchmark-junit4 = "1.2.1"
 androidx-compose-bom = "2024.05.00"
+androidx-compose-material3-adaptive = "1.0.0-beta01"
```
Comment (bentrengrove) [inline Reply/gradle/libs.versions.toml:13]: You need to make all these changes in the scripts/libs.versions.toml file or else they will be overriden on the next dependency update

File: Reply/app/src/main/java/com/example/reply/ui/navigation/ReplyNavigationComponents.kt (hunk 1 above)
Comment (jdkoren) [reply Reply/app/src/main/java/com/example/reply/ui/navigation/ReplyNavigationComponents.kt:118]: I tried something like that, and it looked fine when changing to a layout that isn't a rail. But when changing back to the rail the drawer opens again, and that's a bit jarring.

We can work around that with a `LaunchedEffect` that calls `drawerState.snapTo(Closed)`, but that feels more imperative than declarative.

File: Reply/gradle/libs.versions.toml (hunk 2 above)
Comment (bentrengrove) [reply Reply/gradle/libs.versions.toml:13]: All good! When I first looked it didn't look like they matched but maybe I was wrong

**Review by bentrengrove — APPROVED** [review]
LGTM

**Pull request merged**